arduino/discovery/discovery.go

```diff
@@ -88,6 +90,7 @@ func (d *Discovery) Start() error {
 
 // List retrieve the port list from this discovery
 func (d *Discovery) List() ([]*BoardPort, error) {
+	d.Lock()
 	if _, err := d.in.Write([]byte("LIST\n")); err != nil {
 		return nil, fmt.Errorf("sending LIST command to discovery: %s", err)
```

Comment:
If the method returns here with an error the structure is left locked.
It's preferable to use `defer` to unlock, since it automatically unlocks, no matter what the code flow.

```
{
   d.Lock()
   defer d.Unlock()
   ...do stuff with d...
}
```